nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -627,6 +627,11 @@ public Description matchMethod(MethodTree tree, VisitorState state) {
       Symbol.MethodSymbol closestOverriddenMethod =
           NullabilityUtil.getClosestOverriddenMethod(methodSymbol, state.getTypes());
       if (closestOverriddenMethod != null) {
+        if (config.isJSpecifyMode()) {
+          new GenericsChecks(state, config, this)
+              .checkTypeParameterNullnessForMethodOverriding(
+                  tree, methodSymbol, closestOverriddenMethod);
+        }
```

Comment:
This is probably correct, but, just as a question: why do we need to add special handling both here and _within_ `checkOverriding`/`checkParamOverriding`. Why not just one place? Is the reason that we can report a more meaningful error here, but we then need to adjust the computation of what returns/arguments are `@Nullable` for our more generic overriding checks?

Reply:
The check here ensures that any generic type _parameters_ in the return type and parameter types are identical (invariant) across the overriding and overridden methods (I've added a comment to the source code).  The special handling in `checkOverriding`/`checkParamOverriding` is to compute the correct nullability of the _base_ return type and _base_ parameter types for the case of generics.  Please let me know how if there are ways to make this clearer.